examples/arithmetics/src/language-server/arithmetics-validator.ts

```diff
@@ -4,21 +4,22 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import type { ArithmeticsAstType, Definition, Expression, BinaryExpression, Module, DeclaredParameter, FunctionCall } from './generated/ast.js';
-import type { ArithmeticsServices } from './arithmetics-module.js';
-import type { ValidationChecks, ValidationAcceptor } from 'langium';
-import { isNumberLiteral, isFunctionCall, isBinaryExpression } from './generated/ast.js';
-import { applyOp } from './arithmetics-util.js';
-import { MultiMap } from 'langium';
+import type { ValidationAcceptor, ValidationChecks } from 'langium';
+import { MultiMap, stream } from 'langium';
 import { evalExpression } from './arithmetics-evaluator.js';
+import type { ArithmeticsServices } from './arithmetics-module.js';
+import type { ResolvedFunctionCall } from './arithmetics-util.js';
+import { applyOp, isResolvedFunctionCall } from './arithmetics-util.js';
+import type { ArithmeticsAstType, BinaryExpression, DeclaredParameter, Definition, Expression, FunctionCall, Module } from './generated/ast.js';
+import { isBinaryExpression, isDefinition, isFunctionCall, isNumberLiteral } from './generated/ast.js';
 
 export function registerValidationChecks(services: ArithmeticsServices): void {
     const registry = services.validation.ValidationRegistry;
     const validator = services.validation.ArithmeticsValidator;
     const checks: ValidationChecks<ArithmeticsAstType> = {
         BinaryExpression: validator.checkDivByZero,
-        Definition: validator.checkNormalisable,
-        Module: validator.checkUniqueDefinitions,
+        Definition: [validator.checkUniqueParmeters, validator.checkNormalisable],
```

Comment:
Could you fix the type along the way? `checkUniqueParameters`

Reply:
Fixed that (you probably meant `checkMatchingParameters`)

Reply:
Got it, fixed